avocado/plugins/run.py

```diff
@@ -99,6 +99,12 @@ def configure(self, parser):
                             "system information (hardware details, profilers, "
                             "etc.). Current:  %(default)s")
 
+        parser.add_argument("--suite-order", default="variants-per-test",
+                            choices=("tests-per-variant",
+                                     "variants-per-test"),
+                            help="How to iterate through test suite and "
```

Comment:
I like the `--execution-order` and originally I wanted to name the options `variant-test` but it looked odd. Yours `variant-then-test` seems understandable just I'd probably use `variants-then-tests` but still I'd slightly prefer the `variants-per-test`. @apahim any opinions?